types/src/block_metadata.rs

```diff
@@ -33,6 +33,7 @@ pub struct BlockMetadata {
     round: u64,
     previous_block_votes: Vec<bool>,
     proposer: AccountAddress,
+    failed_proposer_indices: Vec<u32>,
```

Comment:
Is it possible that a proposer failed multiple times? Do we want to take that into account here or that is an overkill?

Reply:
Yeah, same proposer can fail multiple times, and will be multiple time repeated in this list.

list is in the rounds order

Reply:
you can see the example in the recover_on_restart test here :
https://github.com/aptos-labs/aptos-core/pull/1481/commits/b5416f40620af8016e1ce72bb686d9917c2f07be#diff-e635da0173fb15af50f16e92fbeb662e4da79a18ea68cd522e2c32e463cec5c4R743